azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -81,46 +81,22 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
         elif not os.path.exists(trans_prv_file):
             raise IOError(errno.ENOENT, "File not found", trans_prv_file)
         else:
-            first_cmd = [self.openssl_cmd, "cms", "-decrypt", "-in", p7m_file, "-inkey",
-                trans_prv_file, "-recip", trans_cert_file]
-            second_cmd = [self.openssl_cmd, "pkcs12", "-nodes", "-password", "pass:", 
-                "-out", pem_file]
-
-            first_proc = subprocess.Popen(first_cmd, stdout=subprocess.PIPE)
-
-            second_proc = subprocess.Popen(second_cmd, stdin=first_proc.stdout,
-                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            first_proc.stdout.close()  # see https://docs.python.org/2/library/subprocess.html#replacing-shell-pipeline
-            stdout, stderr = second_proc.communicate()
-
-            if second_proc.returncode != 0:
-                stdout = ustr(stdout, encoding='utf-8', errors="backslashreplace") if stdout else ""
-                stderr =  ustr(stderr, encoding='utf-8', errors="backslashreplace") if stderr else ""
-
-                msg = "Failed to decrypt {0}\n[stdout]\n{1}\n\n[stderr]\n{2}\n"\
-                    .format(p7m_file, stdout, stderr)
-                logger.error(msg)
-            
+            try:
+                shellutil.run_pipe([
+                    [self.openssl_cmd, "cms", "-decrypt", "-in", p7m_file, "-inkey", trans_prv_file, "-recip", trans_cert_file],
+                    [self.openssl_cmd, "pkcs12", "-nodes", "-password", "pass:", "-out", pem_file]])
+            except shellutil.CommandError as command_error:
+                logger.error("Failed to decrypt {0} (return code: {1})\n[stdout]\n{2}\n[stderr]\n{3}",
+                    p7m_file, command_error.returncode, command_error.stdout, command_error.stderr)
 
     def crt_to_ssh(self, input_file, output_file):
         with open(output_file, "ab") as file_out:
             cmd = ["ssh-keygen", "-i", "-m", "PKCS8", "-f", input_file]
 
             try:
-                keygen_proc = subprocess.Popen(cmd, stdout=file_out)
-                keygen_proc.wait()
-
-                if keygen_proc.returncode != 0:
-                    msg = u"Command: [{0}], return code: [{1}]" \
-                        .format(cmd, keygen_proc.returncode)
-                    logger.error(msg)
-                
-            except Exception as exception:
-                msg = u"Exception on Command: [{0}]. exception={1}" \
-                    .format(cmd, exception)
-
-
-
+                shellutil.run_command(cmd, stdout=file_out, log_error=True)
+            except shellutil.CommandError:
```

Comment:
Previously, we were catching all exception types here. Would it be safer to keep it that way?

Reply:
The original code read  like this:

```
    def crt_to_ssh(self, input_file, output_file):
        shellutil.run("ssh-keygen -i -m PKCS8 -f {0} >> {1}".format(input_file,
                                                                    output_file))
```
The try/except was introduced in one of the iterations moving to run_command.

Reply:
I see, thanks for confirming.